<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Physcraper Examples: the Hollies • physcraperex</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Physcraper Examples: the Hollies">
<meta property="og:description" content="physcraperex">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">physcraperex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ilex.html">Physcraper Examples: the Hollies</a>
    </li>
    <li>
      <a href="../articles/malvaceae.html">Physcraper Examples: the Malvaceae</a>
    </li>
    <li>
      <a href="../articles/schoch.html">Physcraper Examples: the Dothideomycetes</a>
    </li>
    <li>
      <a href="../articles/todo.html">TODO</a>
    </li>
    <li>
      <a href="../articles/verification-test.html">ilex-physcraper-test</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="ilex_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Physcraper Examples: the Hollies</h1>
            
      
      
      <div class="hidden name"><code>ilex.Rmd</code></div>

    </div>

    
    
<div class="figure">
<img src="ilex-snow.jpg" alt="*Ilex* fruits and leaves in the snow. Image by [Wolfgang Claussen](https://pixabay.com/es/users/WolfBlur-2503887/) at [Pixabay](https://pixabay.com/es/photos/holly-ilex-aquifolium-la-nieve-3012084/)." width="75%"><p class="caption">
<em>Ilex</em> fruits and leaves in the snow. Image by <a href="https://pixabay.com/es/users/WolfBlur-2503887/">Wolfgang Claussen</a> at <a href="https://pixabay.com/es/photos/holly-ilex-aquifolium-la-nieve-3012084/">Pixabay</a>.
</p>
</div>
<div id="i--finding-a-tree-to-update" class="section level2">
<h2 class="hasAnchor">
<a href="#i--finding-a-tree-to-update" class="anchor"></a>I. Finding a tree to update</h2>
<div id="using-the-open-tree-of-life-website" class="section level3">
<h3 class="hasAnchor">
<a href="#using-the-open-tree-of-life-website" class="anchor"></a>Using the Open Tree of Life website</h3>
<p>Go to the <a href="https://tree.opentreeoflife.org/opentree/argus/opentree12.3@ott93302">Open Tree of Life website</a> and use the “search for taxon” menu to look up for the taxon <em>Ilex</em>.</p>
<p>The screenshpt below shows the state of the genus <em>Ilex</em> in the Open Tree of Life synthetic tree on May 25, 2020. If you go to OpenTree’s website you can check out <em>Ilex</em> current state <a href="https://tree.opentreeoflife.org/opentree/argus/ottol@727571/Ilex">here</a>.</p>
<p><img src="ilex-tree-opentree-2020-05-31.png" width="100%"></p>
<hr>
<p>Navigating into the tree, we notice that there might be two studies associated to this portion of the Open Tree synthetic tree.</p>
<p><img src="ilex-tree-opentree-2020-05-31-2.png" width="100%"></p>
<hr>
<p>Let’s verify that on the <a href="https://tree.opentreeoflife.org/curator">study curator of OToL</a>.</p>
<p>Studies matching the word ‘ilex’ on the curator database, at the middle of year 2020. Some of these studies are not actually about the hollies, but other taxa that have the species epithet <em>ilex</em>, e.g., the holly oak <em>Quercus ilex</em> or the rodent <em>Apodemus ilex</em></p>
<p><img src="ilex-curation-opentree-2020-05-31.png" width="100%"></p>
</div>
<div id="using-the-r-package-rotl" class="section level3">
<h3 class="hasAnchor">
<a href="#using-the-r-package-rotl" class="anchor"></a>Using the R package rotl</h3>
<!-- ```{r} -->
<!-- ilex <- rotl::tnrs_match_names(names = "ilex") -->
<!-- ``` -->
<!-- ```{r, results='asis', echo=FALSE} -->
<!-- knitr::kable(ilex, caption = "Taxonomic Name Resolution information for the genus *Ilex*") -->
<!-- ``` -->
<!-- Can we use the *OTT id* to get some trees or studies? -->
<!-- ```{r} -->
<!-- rotl::studies_properties() -->
<!-- ``` -->
<p>Explain what a focal clade is.</p>
<p>There is a handy function that will search a taxon among the focal clades reported across trees.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="va">studies</span> <span class="op">&lt;-</span> <span class="fu">rotl</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rotl/reference/studies_find_studies.html">studies_find_studies</a></span><span class="op">(</span>property<span class="op">=</span><span class="st">"ot:focalCladeOTTTaxonName"</span>, value<span class="op">=</span><span class="st">"Ilex"</span><span class="op">)</span></pre></div>
<table class="table">
<caption>Studies with the genus <em>Ilex</em> as focal clade.</caption>
<colgroup>
<col width="4%">
<col width="3%">
<col width="8%">
<col width="4%">
<col width="5%">
<col width="55%">
<col width="17%">
</colgroup>
<thead><tr class="header">
<th align="left">study_ids</th>
<th align="left">n_trees</th>
<th align="left">tree_ids</th>
<th align="left">candidate</th>
<th align="left">study_year</th>
<th align="left">title</th>
<th align="left">study_doi</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ot_1984</td>
<td align="left">1</td>
<td align="left">tree1</td>
<td align="left"></td>
<td align="left">2020</td>
<td align="left"></td>
<td align="left"><a href="http://dx.doi.org/10.1111/jse.12567" class="uri">http://dx.doi.org/10.1111/jse.12567</a></td>
</tr>
<tr class="even">
<td align="left">pg_2827</td>
<td align="left">2</td>
<td align="left">tree6576, tree6577</td>
<td align="left">tree6577</td>
<td align="left">2005</td>
<td align="left">Molecular analyses of the genus Ilex (Aquifoliaceae) in southern South America, evidence from AFLP and ITS sequence data</td>
<td align="left"><a href="http://dx.doi.org/10.3732/ajb.92.2.352" class="uri">http://dx.doi.org/10.3732/ajb.92.2.352</a></td>
</tr>
</tbody>
</table>
<p>It seems like the oldest tree, <em>tree6577</em> from study <em>pg_2827</em>, is in the Open Tree of Life synthetic tree.</p>
<p>Let’s get it and plot it here:</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="va">original_tree</span> <span class="op">&lt;-</span> <span class="fu">rotl</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rotl/reference/get_study_tree.html">get_study_tree</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"pg_2827"</span>, tree_id <span class="op">=</span> <span class="st">"tree6577"</span><span class="op">)</span>
<span class="co">#&gt; Warning in build_raw_phylo(ncl, missing_edge_length): missing edge lengths are</span>
<span class="co">#&gt; not allowed in phylo class. All removed.</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/plot.phylo.html">plot.phylo</a></span><span class="op">(</span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/ladderize.html">ladderize</a></span><span class="op">(</span><span class="va">original_tree</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"phylogram"</span>, cex <span class="op">=</span> <span class="fl">0.3</span>, label.offset <span class="op">=</span> <span class="fl">1</span>, edge.width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></pre></div>
<p><img src="ilex_files/figure-html/original-tree-1.png" width="100%" style="background-color: #DCDCDC; padding:10px; display: inline-block;"></p>
<hr>
<p>Now, let’s look at some properties of the tree:</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Ntip</a></span><span class="op">(</span><span class="va">original_tree</span><span class="op">)</span>  <span class="co"># gets the number of tips</span>
<span class="co">#&gt; [1] 48</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/root.html">is.rooted</a></span><span class="op">(</span><span class="va">original_tree</span><span class="op">)</span>  <span class="co"># check that it is rooted</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/is.binary.tree.html">is.binary</a></span><span class="op">(</span><span class="va">original_tree</span><span class="op">)</span>  <span class="co"># check that it is fully resolved</span>
<span class="co">#&gt; [1] FALSE</span>
<span class="fu">datelife</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/datelife/man/phylo_has_brlen.html">phylo_has_brlen</a></span><span class="op">(</span><span class="va">original_tree</span><span class="op">)</span>  <span class="co"># checks that it has branch lengths</span>
<span class="co">#&gt; [1] FALSE</span></pre></div>
<p>The tree has 48 tips, is rooted, has no branch lengths and is not fully resolved, as you probably already noted. Also, labels correspond to the labels reported on the original study <a href="http://dx.doi.org/10.3732/ajb.92.2.352">here</a>. Other labels are available to use as tip labels. For example, you can plot the tree using the unified taxonomic names, or the taxonomic ids.</p>
</div>
<div id="finding-a-tree-in-the-open-tree-of-life-phylesystem-with-at-least-one-tip-label-belonging-to-the-group-of-interest-" class="section level3">
<h3 class="hasAnchor">
<a href="#finding-a-tree-in-the-open-tree-of-life-phylesystem-with-at-least-one-tip-label-belonging-to-the-group-of-interest-" class="anchor"></a>Finding a tree in the Open Tree of Life phylesystem with at least one tip label belonging to the group of interest.</h3>
<pre><code>find_trees.py --taxon_name  "Ilex"</code></pre>
<hr>
</div>
</div>
<div id="ii--getting-the-underlying-alignment" class="section level2">
<h2 class="hasAnchor">
<a href="#ii--getting-the-underlying-alignment" class="anchor"></a>II. Getting the underlying alignment</h2>
<div id="from-treebase" class="section level3">
<h3 class="hasAnchor">
<a href="#from-treebase" class="anchor"></a>From TreeBASE</h3>
<div id="using-physcraper-and-the-arguments--tb-and--no_est" class="section level4">
<h4 class="hasAnchor">
<a href="#using-physcraper-and-the-arguments--tb-and--no_est" class="anchor"></a>Using physcraper and the arguments <code>-tb</code> and <code>-no_est</code>
</h4>
<pre><code>physcraper_run.py -s pg_2827 -t tree6577 -tb -no_est -o data/pg_2827_tree6577</code></pre>
</div>
<div id="downloading-the-alignment-directly-from-a-repository" class="section level4">
<h4 class="hasAnchor">
<a href="#downloading-the-alignment-directly-from-a-repository" class="anchor"></a>Downloading the alignment directly from a repository</h4>
<p>The alignments are here <a href="https://treebase.org/treebase-web/search/study/matrices.html?id=1091" class="uri">https://treebase.org/treebase-web/search/study/matrices.html?id=1091</a></p>
<p>On a mac you can do:</p>
<pre><code>wget "http://purl.org/phylo/treebase/phylows/matrix/TB2:M2478?format=nexus" -o data-raw/alignments/T1281-M2478.nex</code></pre>
</div>
</div>
<div id="other-repos" class="section level3">
<h3 class="hasAnchor">
<a href="#other-repos" class="anchor"></a>Other repos</h3>
<hr>
</div>
</div>
<div id="iii--starting-a-physcraper-a-run" class="section level2">
<h2 class="hasAnchor">
<a href="#iii--starting-a-physcraper-a-run" class="anchor"></a>III. Starting a Physcraper a run</h2>
<pre><code>physcraper_run.py -s pg_2827 -t tree6577 -o data/pg_2827_tree6577</code></pre>
<pre><code>physcraper_run.py -s pg_2827 -t tree6577 -a data-raw/alignments/T1281-M2478.nex -as nexus -o data/ilex-remote</code></pre>
<div id="using-a-local-blast-database" class="section level3">
<h3 class="hasAnchor">
<a href="#using-a-local-blast-database" class="anchor"></a>Using a local BLAST database</h3>
<pre><code>physcraper_run.py -s pg_2827 -t tree6577 -a data-raw/alignments/T1281-M2478.nex -as nexus -db local_blast_db -o data/ilex-local</code></pre>
<hr>
</div>
</div>
<div id="iv--reading-the-physcraper-results" class="section level2">
<h2 class="hasAnchor">
<a href="#iv--reading-the-physcraper-results" class="anchor"></a>IV. Reading the Physcraper results</h2>
<div id="the-pxhyscraper-tag" class="section level3">
<h3 class="hasAnchor">
<a href="#the-pxhyscraper-tag" class="anchor"></a>The Pxhyscraper tag</h3>
<p>T1281-M2478</p>
</div>
<div id="input-files" class="section level3">
<h3 class="hasAnchor">
<a href="#input-files" class="anchor"></a>Input files</h3>
<p>Physcraper rewrites input files for a couple reasons: reproducibility, taxon name matching, and taxon reconciliation. It also writes the config file down if none was provided.</p>
</div>
<div id="run-files" class="section level3">
<h3 class="hasAnchor">
<a href="#run-files" class="anchor"></a>Run files</h3>
<p>Files in here are also automatically written down by Physcraper.</p>
<p>blast runs, alignments, raxml trees, bootstrap</p>
<p>The trees are reconstructed using RAxML, with tip labels corresponding to Physcraper taxon ids (e.g., otu42009, otuPS1) and not taxon names (e.g., <em>Helwingia japonica</em>), nor taxonomic ids (e.g., ott or ncbi). Branch lengths are proportional to relative substitution rates. The RAxML tree with taxon names as tip labels is saved on the <code>outputs_tag</code> folder.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="va">updated_tree_otus</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"../data/ilex-local/run_T1281-M2478/RAxML_bestTree.2020-06-29"</span><span class="op">)</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/plot.phylo.html">plot.phylo</a></span><span class="op">(</span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/ladderize.html">ladderize</a></span><span class="op">(</span><span class="va">updated_tree_otus</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"fan"</span>, cex <span class="op">=</span> <span class="fl">0.25</span>, label.offset <span class="op">=</span> <span class="fl">0.01</span>, edge.width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/add.scale.bar.html">add.scale.bar</a></span><span class="op">(</span>cex <span class="op">=</span> <span class="fl">0.3</span>, font <span class="op">=</span> <span class="fl">1</span>, col <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span><span class="st">"Updated tree - otu tags as labels"</span>, side <span class="op">=</span> <span class="fl">3</span>, line <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></pre></div>
<p><img src="ilex_files/figure-html/updated-tree-otus-1.png" width="100%" style="background-color: #6B8E23; padding:10px; display: inline-block;"></p>
</div>
<div id="output-files" class="section level3">
<h3 class="hasAnchor">
<a href="#output-files" class="anchor"></a>Output files</h3>
<p>A variety of files are automatically written down by Physcraper.</p>
<p>A nexson tree with all types of tip labels is saved in here. From this tree, a tree with any kind of label can be produced. By default, the updated tree with taxon names as tip labels is saved in the <code>output_tag</code> folder as <code>updated_taxonname.tre</code>.</p>
<p>Updated_sumtree and updated_tree_taxonname are the same tree: a consensus tree output of <a href="https://dendropy.org/programs/sumtrees.html">summtrees from DendroPy</a>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="va">updated_sumtree</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"../data/pg_2827_tree6577/outputs_pg_2827tree6577/physcraper_pg_2827tree6577.tre"</span><span class="op">)</span>

<span class="va">updated_tree_taxonname</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span>
  file <span class="op">=</span> <span class="st">"../data/pg_2827_tree6577/outputs_pg_2827tree6577/updated_taxonname.tre"</span><span class="op">)</span>
<span class="co"># ape::plot.phylo(ape::ladderize(updated_tree_taxonname), cex = 0.35)</span>
<span class="co"># updated_tree_taxonname$tip.label %in% c("Helwingia_japonica_otu420083", "Helwingia_chinensis_otu420099")</span>

<span class="co"># updated_raxml &lt;- ape::read.tree(file = "../data/pg_2827_tree6577/run_pg_2827tree6577_run4/RAxML_bestTree.2020-07-31")</span>
<span class="va">updated_sumtree_drop</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/drop.tip.html">drop.tip</a></span><span class="op">(</span><span class="va">updated_sumtree</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"otu420083"</span>, <span class="st">"otu420099"</span><span class="op">)</span><span class="op">)</span></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/plot.phylo.html">plot.phylo</a></span><span class="op">(</span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/ladderize.html">ladderize</a></span><span class="op">(</span><span class="va">updated_tree_taxonname</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"fan"</span>, cex <span class="op">=</span> <span class="fl">0.25</span>, label.offset <span class="op">=</span> <span class="fl">0.01</span>, edge.width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/add.scale.bar.html">add.scale.bar</a></span><span class="op">(</span>cex <span class="op">=</span> <span class="fl">0.3</span>, font <span class="op">=</span> <span class="fl">1</span>, col <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span><span class="st">"Updated tree - Taxon names as labels  "</span>, side <span class="op">=</span> <span class="fl">3</span>, line <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></pre></div>
<p><img src="ilex_files/figure-html/updated-tree-taxonname-1.png" width="100%" style="background-color: #9ecff7; padding:10px; display: inline-block;"></p>
<hr>
</div>
</div>
<div id="v--visualizing-the-physcraper-results" class="section level2">
<h2 class="hasAnchor">
<a href="#v--visualizing-the-physcraper-results" class="anchor"></a>V. Visualizing the Physcraper results</h2>
<pre><code>#&gt; [1] 61
#&gt; [1] 30
#&gt; [1] 77
#&gt; [1] 207
#&gt; [1] 1.000001e-06 1.000001e-06 1.000001e-06 1.000001e-06 1.000001e-06
#&gt; [6] 1.000001e-06 1.000001e-06 2.000001e-06
#&gt; [1] 0.96 0.95 0.95 0.93 0.76 0.76 0.93 0.86
#&gt; [1] 6.716139e-03 1.982183e-01 1.130793e-02 1.000001e-06 8.803937e-03
#&gt; [6] 1.000001e-06 1.138649e-02 7.447242e-03</code></pre>
<div id="visualizing-new-tips-and-taxa" class="section level3">
<h3 class="hasAnchor">
<a href="#visualizing-new-tips-and-taxa" class="anchor"></a><em>Visualizing new tips and taxa</em>
</h3>
<p>An easy way to visualize the new tips added into the updated tree compared to the original tree is coloring the tips and branches from those new tips with a highlighting color. We will use the function <code>plot_branches</code> defined in this package to do so.</p>
<p>The function takes as argument the paths to both the tree and the OTU info files:</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="co"># mytreefile = '../data/pg_2827_tree6577/run_pg_2827tree6577_run4/RAxML_bestTree.2020-07-31'</span>
<span class="va">mytreefile</span> <span class="op">=</span> <span class="st">'../data/pg_2827_tree6577/outputs_pg_2827tree6577/physcraper_pg_2827tree6577.tre'</span>
<span class="va">myotuinfofile</span> <span class="op">=</span> <span class="st">'../data/pg_2827_tree6577/outputs_pg_2827tree6577/otu_info_pg_2827tree6577.csv'</span></pre></div>
<p>And uses <code><a href="https://rdrr.io/pkg/phytools/man/plotBranchbyTrait.html">phytools::plotBranchbyTrait</a></code>, so it takes all arguments from <code><a href="https://rdrr.io/pkg/ape/man/plot.phylo.html">ape::plot.phylo</a></code>, too:</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="fu"><a href="../reference/plot_branches.html">plot_branches</a></span><span class="op">(</span>treefile <span class="op">=</span> <span class="va">mytreefile</span>, otufile <span class="op">=</span> <span class="va">myotuinfofile</span>, type <span class="op">=</span> <span class="st">"fan"</span>, cex <span class="op">=</span> <span class="fl">0.25</span>, edge.width <span class="op">=</span> <span class="fl">0.5</span>, tip_label <span class="op">=</span> <span class="st">"taxon"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/expression.html">expression</a></span><span class="op">(</span><span class="va">Updated</span><span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">italic</a></span><span class="op">(</span><span class="va">Ilex</span><span class="op">)</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/plotmath.html">plain</a></span><span class="op">(</span><span class="st">"gene tree - 4 BLAST cycles"</span><span class="op">)</span><span class="op">)</span>, side<span class="op">=</span> <span class="fl">3</span>, line<span class="op">=</span><span class="op">-</span><span class="fl">1</span>, cex <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/nodelabels.html">nodelabels</a></span><span class="op">(</span>node<span class="op">=</span><span class="va">bb</span><span class="op">+</span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Ntip</a></span><span class="op">(</span><span class="va">updated_sumtree_drop</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span>, pch <span class="op">=</span><span class="fl">8</span>, cex <span class="op">=</span> <span class="fl">0.1</span>, col <span class="op">=</span> <span class="st">"green"</span><span class="op">)</span></pre></div>
<p><img src="ilex_files/figure-html/plot-tips-and-branches-1.png" width="100%" style="background-color: #DCDCDC; padding:10px; display: inline-block;"></p>
<p>The green nodes highlight branches with a length &lt; 0.00001 and a bootstrap support &gt; 0.75</p>
<p>Let’s get the summary of new tips:</p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="va">summ</span> <span class="op">&lt;-</span> <span class="fu">summarize</span><span class="op">(</span>treefile <span class="op">=</span> <span class="va">mytreefile</span>, otufile <span class="op">=</span> <span class="va">myotuinfofile</span><span class="op">)</span>
<span class="co">#&gt; This updated tree has 48 original tips, representing 45 unique OTT taxa.</span>
<span class="co">#&gt; And 231 new tips representing 83 new OTT taxa added to the original tree.</span>
<span class="co">#&gt; It is 0.75% resolved.</span>
<span class="va">updated_sumtree_drop</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/ladderize.html">ladderize</a></span><span class="op">(</span><span class="va">updated_sumtree_drop</span><span class="op">)</span>
<span class="va">internal_edges</span> <span class="op">&lt;-</span> <span class="va">updated_sumtree_drop</span><span class="op">$</span><span class="va">edge</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">&gt;</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Ntip</a></span><span class="op">(</span><span class="va">updated_sumtree_drop</span><span class="op">)</span>
<span class="va">tip_edges</span> <span class="op">&lt;-</span> <span class="va">updated_sumtree_drop</span><span class="op">$</span><span class="va">edge</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span> <span class="op">&lt;=</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Ntip</a></span><span class="op">(</span><span class="va">updated_sumtree_drop</span><span class="op">)</span>
<span class="va">boots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span><span class="op">(</span><span class="va">updated_sumtree_drop</span><span class="op">$</span><span class="va">node.label</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">boots</span><span class="op">&gt;</span><span class="fl">0.8</span><span class="op">)</span>
<span class="co">#&gt; [1] 61</span>
<span class="co"># internal edges that are very small:</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">updated_sumtree_drop</span><span class="op">$</span><span class="va">edge.length</span><span class="op">[</span><span class="va">internal_edges</span><span class="op">]</span><span class="op">&lt;</span><span class="fl">0.00001</span><span class="op">)</span> <span class="co"># 30</span>
<span class="co">#&gt; [1] 30</span>
<span class="co"># tip edges that are very small:</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">updated_sumtree_drop</span><span class="op">$</span><span class="va">edge.length</span><span class="op">[</span><span class="va">tip_edges</span><span class="op">]</span><span class="op">&lt;</span><span class="fl">0.00001</span><span class="op">)</span> <span class="co"># 77</span>
<span class="co">#&gt; [1] 77</span>
<span class="va">updated_sumtree_drop</span><span class="op">$</span><span class="va">Nnode</span>
<span class="co">#&gt; [1] 207</span>
<span class="va">ie</span> <span class="op">&lt;-</span> <span class="va">updated_sumtree_drop</span><span class="op">$</span><span class="va">edge.length</span><span class="op">[</span><span class="va">internal_edges</span><span class="op">]</span>
<span class="va">ie</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">ie</span><span class="op">)</span> <span class="co"># adding a root edge length</span>
<span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span><span class="op">(</span><span class="va">boots</span><span class="op">&lt;</span><span class="fl">0.75</span><span class="op">)</span><span class="op">/</span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Ntip</a></span><span class="op">(</span><span class="va">updated_sumtree_drop</span><span class="op">)</span>
<span class="co">#&gt; [1] 0.4873646</span>
<span class="va">bb</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">ie</span><span class="op">&lt;</span><span class="fl">0.00001</span> <span class="op">&amp;</span> <span class="va">boots</span><span class="op">&gt;</span><span class="fl">0.75</span><span class="op">)</span>
<span class="va">ie</span><span class="op">[</span><span class="va">bb</span><span class="op">]</span>
<span class="co">#&gt; [1] 1.000001e-06 1.000001e-06 1.000001e-06 1.000001e-06 1.000001e-06</span>
<span class="co">#&gt; [6] 1.000001e-06 1.000001e-06 2.000001e-06</span>
<span class="va">boots</span><span class="op">[</span><span class="va">bb</span><span class="op">]</span>
<span class="co">#&gt; [1] 0.96 0.95 0.95 0.93 0.76 0.76 0.93 0.86</span>
<span class="va">updated_sumtree_drop</span><span class="op">$</span><span class="va">edge.length</span><span class="op">[</span><span class="va">bb</span><span class="op">+</span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Ntip</a></span><span class="op">(</span><span class="va">updated_sumtree_drop</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt; [1] 6.716139e-03 1.982183e-01 1.130793e-02 1.000001e-06 8.803937e-03</span>
<span class="co">#&gt; [6] 1.000001e-06 1.138649e-02 7.447242e-03</span>
<span class="va">ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">bb</span><span class="op">+</span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Ntip</a></span><span class="op">(</span><span class="va">updated_sumtree_drop</span><span class="op">)</span><span class="op">-</span><span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span>
              <span class="fu">phytools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phytools/man/getDescendants.html">getDescendants</a></span><span class="op">(</span>tree<span class="op">=</span><span class="va">updated_sumtree_drop</span>, node<span class="op">=</span><span class="va">x</span><span class="op">)</span><span class="op">)</span></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">updated_sumtree_drop</span><span class="op">$</span><span class="va">edge.length</span>, breaks<span class="op">=</span><span class="fl">1400</span>, col <span class="op">=</span> <span class="st">"skyblue3"</span>,
     border <span class="op">=</span> <span class="st">"skyblue3"</span>, xlab<span class="op">=</span> <span class="st">"Branch Length (substitution rate)"</span>,
     main <span class="op">=</span> <span class="st">"Updated Gottlieb2005 tree \n branch length frequency distribution"</span><span class="op">)</span></pre></div>
<p><img src="ilex_files/figure-html/hist-1.png" width="100%" style="background-color: #DCDCDC; padding:10px; display: inline-block;"></p>
<p>This branch length distribution excludes the outgroup branches, it contains branches from the ingroup only.</p>
<p>The updated tree is 75% resolved, with 249 branches that are &lt;0.01 and 107 are &lt; 0.00001, effectively negligible. The longest branch is 0.2416085 and the smallest branch is 1.000000510^{-6}</p>
<hr>
</div>
</div>
<div id="v--comparing-the-physcraper-results" class="section level2">
<h2 class="hasAnchor">
<a href="#v--comparing-the-physcraper-results" class="anchor"></a>V. Comparing the Physcraper results</h2>
<div id="visualizing-changes-in-phylogenetic-relationships" class="section level3">
<h3 class="hasAnchor">
<a href="#visualizing-changes-in-phylogenetic-relationships" class="anchor"></a><em>Visualizing changes in phylogenetic relationships</em>
</h3>
<p>How do the phylogenetic relationships from the updated tree compare to those from the original tree and other <em>Ilex</em> trees?</p>
<p>We will use the conflict tool of the OpenTree to visualize this.</p>
<p>The next figure shows in green the nodes resolved by the Gottlieb2005 tree and in blue the nodes that align with taxonomy: <img src="ilex-tree6577-conflict-vs-taxonomy.png" width="100%"></p>
<p>The following figure shows results from the conflict analysis in the updated <em>Ilex</em> tree. The blue nodes either align with phylogenetic relationships from the Gottlieb2005 tree or with taxonomy; the green nodes (approximately 20) resolve phylogenetic relationships that were not solved by Gottlieb2005; and the orange nodes conflict either with phylogenetic relationships shown in Gottlieb2005 or with taxonomy.</p>
<p><img src="ilex-tree6577-physcraper-conflict-vs-synth.png" width="100%"></p>
<p>Let’s get the Yao2020 tree:</p>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="va">yao2020</span> <span class="op">&lt;-</span> <span class="fu">rotl</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rotl/reference/get_study_tree.html">get_study_tree</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"ot_1984"</span>, tree_id <span class="op">=</span> <span class="st">"tree1"</span><span class="op">)</span></pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span><span class="op">(</span><span class="va">yao2020</span><span class="op">$</span><span class="va">edge.length</span>, col <span class="op">=</span> <span class="st">"skyblue3"</span>, breaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span><span class="op">(</span><span class="va">yao2020</span><span class="op">$</span><span class="va">edge.length</span><span class="op">)</span><span class="op">)</span>,
     border <span class="op">=</span> <span class="st">"skyblue3"</span>, xlab<span class="op">=</span> <span class="st">"Branch Length (million years)"</span>,
     main <span class="op">=</span> <span class="st">"Yao2020 tree branch length \n frequency distribution"</span><span class="op">)</span></pre></div>
<p><img src="ilex_files/figure-html/hist2-1.png" width="100%" style="background-color: #DCDCDC; padding:10px; display: inline-block;"></p>
<p>Now we can compare the taxa in the updated Gottlieb2005 and in Yao2020:</p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="va">comp</span> <span class="op">&lt;-</span> <span class="fu">compare</span><span class="op">(</span>treefile <span class="op">=</span> <span class="va">mytreefile</span>, otufile <span class="op">=</span> <span class="va">myotuinfofile</span>, tree <span class="op">=</span> <span class="va">yao2020</span><span class="op">)</span>
<span class="co">#&gt; The updated tree (from treefile) and the tree to compare (from tree)</span>
<span class="co">#&gt;   contribute jointly with 151 distinct taxa that are in not in the original tree.</span>
<span class="co">#&gt;  From this, 65 taxa are present in both trees.</span>
<span class="co">#&gt; The updated tree contributes 18 distinct taxa that are neither in original tree</span>
<span class="co">#&gt;   nor in tree to compare.</span>
<span class="co">#&gt; The tree to compare has 133 new tips and contributes 133 distinct taxa </span>
<span class="co">#&gt;  from which 68 are neither in original tree nor in the updated tree.</span>
<span class="co">#&gt; The tree to compare has 42 taxa that are in original tree.</span>
<span class="co">#&gt; And 133 tips different from original, representing 68 new OTT taxa added to the original tree.</span>
<span class="co">#&gt; It is 0.99% resolved.</span></pre></div>
<p>The following figure shows results from the conflict analysis on the Yao2020 tree. Again, the blue nodes align either with Gottlieb2005 or with taxonomy, the green nodes (approximately 80) resolve phylogenetic relationships that were not solved by Gottlieb2005, and the orange nodes conflict either with phylogenetic relationships from Gottlieb2005 or with taxonomy.</p>
<p><img src="ilex-tree1-physcraper-conflict-vs-synth.png" width="100%"></p>
<p>With Python:</p>
<pre><code>from opentree import OT
import physcraper

gottleib_updated = physcraper.treetaxon.generate_TreeTax_from_run("../physcraperex/data/ilex-local/")
gottleib_updated.write_labelled(path = "gott_taxonlab.tre",label = '^ot:ottTaxonName', norepeats=False)</code></pre>
<p>The following generates a tree with tip labels with spaces instead of underscores.</p>
<pre><code>yao = OT.get_tree(study_id="ot_1984", tree_id="tree1",tree_format = "newick", label_format='ot:ottTaxonName')
fi = open("yao.tre", "w")
fi.write(yao.response_dict['content'].decode())
fi.close() </code></pre>
<p>Use the tree comparison script:</p>
<pre><code>tree_comparison.py -t1 gott_taxonlab.tre -t2 yao.tre -og otu468906 otu1030991 -otu data/pg_2827_tree6577/outputs_pg_2827tree6577/otu_info_pg_2827tree6577.csv -o ilex_comparison
</code></pre>
<hr>
<hr>
</div>
</div>
<div id="acknowledgments" class="section level2">
<h2 class="hasAnchor">
<a href="#acknowledgments" class="anchor"></a>Acknowledgments</h2>
<p>Research was supported by the grant “Sustaining the Open Tree of Life”, National Science Foundation ABI No. 1759838, and ABI No. 1759846.</p>
<p>Computer time was provided by the Multi-Environment Research Computer for Exploration and Discovery (MERCED) cluster from the University of California, Merced (UCM), supported by the NSF Grant No. ACI-1429783.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Luna L. Sanchez Reyes, Emily Jane McTavish.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
