<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Physcraper Examples: the Malvaceae • physcraperex</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Physcraper Examples: the Malvaceae">
<meta property="og:description" content="physcraperex">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">physcraperex</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/ilex.html">Physcraper Examples: the Hollies</a>
    </li>
    <li>
      <a href="../articles/malvaceae.html">Physcraper Examples: the Malvaceae</a>
    </li>
    <li>
      <a href="../articles/schoch.html">Physcraper Examples: the Dothideomycetes</a>
    </li>
    <li>
      <a href="../articles/todo.html">TODO</a>
    </li>
    <li>
      <a href="../articles/verification-test.html">ilex-physcraper-test</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="malvaceae_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Physcraper Examples: the Malvaceae</h1>
            
      
      
      <div class="hidden name"><code>malvaceae.Rmd</code></div>

    </div>

    
    
<pre><code>#&gt; [1] "/Users/luna/pj_physcraper/physcraperex/vignettes"</code></pre>
<div class="figure">
<img src="malvaceae-plant.jpg" alt="Cocoa plant fruits" width="75%"><p class="caption">
Cocoa plant fruits
</p>
</div>
<div id="i--finding-a-tree-to-update" class="section level2">
<h2 class="hasAnchor">
<a href="#i--finding-a-tree-to-update" class="anchor"></a>I. Finding a tree to update</h2>
<div id="with-the-open-tree-of-life-website" class="section level3">
<h3 class="hasAnchor">
<a href="#with-the-open-tree-of-life-website" class="anchor"></a>With the Open Tree of Life website</h3>
<p>Go to the <a href="https://tree.opentreeoflife.org/opentree/argus/opentree12.3@ott93302">Open Tree of Life website</a> and use the “search for taxon” menu to look up the taxon <strong>Malvaceae</strong>.</p>
<p>This is how the genus <strong>Malvaceae</strong> is represented on the Open Tree of Life synthetic tree at the middle of year 2020:</p>
<div class="figure">
<img src="malvaceae-tree-opentree.png" alt="See this live on the website at &lt;https://tree.opentreeoflife.org/opentree/argus/ottol@279960/Malvaceae&gt;" width="100%"><p class="caption">
See this live on the website at <a href="https://tree.opentreeoflife.org/opentree/argus/ottol@279960/Malvaceae" class="uri">https://tree.opentreeoflife.org/opentree/argus/ottol@279960/Malvaceae</a>
</p>
</div>
<hr>
<p>Let’s go to the <a href="https://tree.opentreeoflife.org/curator">study curator of OToL</a> and look up for uploaded studies for the <strong>Malvaceae</strong></p>
<p>Studies matching the word ‘malvaceae’ on the curator database, at the middle of year 2020. Some of these studies are not actually about the Malvaceae, but other taxa that have the keyword <em>malvaceae</em>.</p>
<div class="figure">
<img src="malvaceae-curation-opentree.png" alt="See this live on the curator app at &lt;https://tree.opentreeoflife.org/curator/?match=malvaceae&gt;" width="100%"><p class="caption">
See this live on the curator app at <a href="https://tree.opentreeoflife.org/curator/?match=malvaceae" class="uri">https://tree.opentreeoflife.org/curator/?match=malvaceae</a>
</p>
</div>
<hr>
</div>
<div id="finding-a-tree-to-update-using-the-r-package-rotl" class="section level3">
<h3 class="hasAnchor">
<a href="#finding-a-tree-to-update-using-the-r-package-rotl" class="anchor"></a>Finding a tree to update using the R package rotl</h3>
<p>Explain what a focal clade is.</p>
<p>There is a handy function that will search a taxon among the focal clades reported across trees.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="va">studies</span> <span class="op">&lt;-</span> <span class="fu">rotl</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rotl/reference/studies_find_studies.html">studies_find_studies</a></span><span class="op">(</span>property<span class="op">=</span><span class="st">"ot:focalCladeOTTTaxonName"</span>, value<span class="op">=</span><span class="st">"Malvaceae"</span><span class="op">)</span></pre></div>
<table class="table">
<caption>Studies with the genus <em>Malvaceae</em> as focal clade.</caption>
<colgroup>
<col width="7%">
<col width="6%">
<col width="30%">
<col width="7%">
<col width="8%">
<col width="4%">
<col width="34%">
</colgroup>
<thead><tr class="header">
<th align="left">study_ids</th>
<th align="left">n_trees</th>
<th align="left">tree_ids</th>
<th align="left">candidate</th>
<th align="left">study_year</th>
<th align="left">title</th>
<th align="left">study_doi</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">pg_628</td>
<td align="left">3</td>
<td align="left">tree1029, tree1030, tree5358</td>
<td align="left"></td>
<td align="left">2011</td>
<td align="left"></td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">pg_629</td>
<td align="left">1</td>
<td align="left">tree1031</td>
<td align="left"></td>
<td align="left">2010</td>
<td align="left"></td>
<td align="left"><a href="http://dx.doi.org/10.1600/036364411X553216" class="uri">http://dx.doi.org/10.1600/036364411X553216</a></td>
</tr>
<tr class="odd">
<td align="left">pg_1855</td>
<td align="left">4</td>
<td align="left">tree3745, tree3746, tree3747, tree3748</td>
<td align="left"></td>
<td align="left">2007</td>
<td align="left"></td>
<td align="left"><a href="http://dx.doi.org/10.1600/036364407780360157" class="uri">http://dx.doi.org/10.1600/036364407780360157</a></td>
</tr>
</tbody>
</table>
<p>None of these trees are in the synthetic tree. Still, the family does not appear as a politomy and has phylogenetic information in the Open Tree of Life synthetic tree. Where is this information coming from?</p>
<p>It is probably coming from studies with focal clades that might be either above or below the particular clade we are interested in!</p>
<p>You can get supporting trees with the Open Tree of Life API, and you can currently do it now using a hidden <code>rotl</code> function:</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">rotl</span><span class="fu">:::</span><span class="fu">.tol_subtree</span><span class="op">(</span>ott_id<span class="op">=</span><span class="fl">279960</span><span class="op">)</span>
<span class="co"># studies &lt;- rotl:::studies_from_otl(res) # not supported anymore as of April 2021</span>
<span class="co"># studies</span></pre></div>
</div>
</div>
<div id="updating-study-pg_55" class="section level1">
<h1 class="hasAnchor">
<a href="#updating-study-pg_55" class="anchor"></a>Updating study <code>pg_55</code>
</h1>
<div class="figure">
<img src="pg_55-curation-opentree.png" alt="See this live on the curator app at &lt;https://tree.opentreeoflife.org/curator/study/view/pg_55&gt;" width="100%"><p class="caption">
See this live on the curator app at <a href="https://tree.opentreeoflife.org/curator/study/view/pg_55" class="uri">https://tree.opentreeoflife.org/curator/study/view/pg_55</a>
</p>
</div>
<!-- Get study data and tree id with rotl -->
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="va">pg_55_meta</span> <span class="op">=</span> <span class="fu">rotl</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rotl/reference/get_study_meta.html">get_study_meta</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"pg_55"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/ls.html">ls</a></span><span class="op">(</span><span class="va">pg_55_meta</span><span class="op">$</span><span class="va">nexml</span><span class="op">)</span>
<span class="va">pg_55_meta</span><span class="op">$</span><span class="va">nexml</span><span class="op">$</span><span class="va">treesById</span>
<span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">pg_55_meta</span><span class="op">$</span><span class="va">nexml</span><span class="op">$</span><span class="va">treesById</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span>
<span class="co"># rotl::taxonomy_taxon_info(pg_55_meta$nexml$"^ot:focalClade")</span>
<span class="co">#&gt;  [1] "@generator"                    "@id"                          </span>
<span class="co">#&gt;  [3] "@nexml2json"                   "@nexmljson"                   </span>
<span class="co">#&gt;  [5] "@version"                      "@xmlns"                       </span>
<span class="co">#&gt;  [7] "^ot:agents"                    "^ot:annotationEvents"         </span>
<span class="co">#&gt;  [9] "^ot:candidateTreeForSynthesis" "^ot:comment"                  </span>
<span class="co">#&gt; [11] "^ot:curatorName"               "^ot:dataDeposit"              </span>
<span class="co">#&gt; [13] "^ot:focalClade"                "^ot:focalCladeOTTTaxonName"   </span>
<span class="co">#&gt; [15] "^ot:otusElementOrder"          "^ot:studyId"                  </span>
<span class="co">#&gt; [17] "^ot:studyPublication"          "^ot:studyPublicationReference"</span>
<span class="co">#&gt; [19] "^ot:studyYear"                 "^ot:tag"                      </span>
<span class="co">#&gt; [21] "^ot:treesElementOrder"         "otusById"                     </span>
<span class="co">#&gt; [23] "treesById"                    </span>
<span class="co">#&gt; $trees55</span>
<span class="co">#&gt; $trees55$treeById</span>
<span class="co">#&gt; $trees55$treeById$tree5864</span>
<span class="co">#&gt; NULL</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $trees55$`^ot:treeElementOrder`</span>
<span class="co">#&gt; $trees55$`^ot:treeElementOrder`[[1]]</span>
<span class="co">#&gt; [1] "tree5864"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $trees55$`@otus`</span>
<span class="co">#&gt; [1] "otus55"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; trees55.^ot:treeElementOrder </span>
<span class="co">#&gt;                   "tree5864"</span></pre></div>
<hr>
<p>The focal clade of study is <strong>Sterculioideae.</strong> Is this taxon in the synthetic tree? FALSE</p>
<p>Let’s get the original tree and plot it here:</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="va">original_tree</span> <span class="op">&lt;-</span> <span class="fu">rotl</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rotl/reference/get_study_tree.html">get_study_tree</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"pg_55"</span>, tree_id <span class="op">=</span> <span class="st">"tree5864"</span><span class="op">)</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/plot.phylo.html">plot.phylo</a></span><span class="op">(</span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/ladderize.html">ladderize</a></span><span class="op">(</span><span class="va">original_tree</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"phylogram"</span>, cex <span class="op">=</span> <span class="fl">0.3</span>, label.offset <span class="op">=</span> <span class="fl">1</span>, edge.width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span><span class="st">"Original tree - original labels"</span>, side <span class="op">=</span> <span class="fl">3</span>, cex <span class="op">=</span> <span class="fl">0.6</span><span class="op">)</span></pre></div>
<p><img src="../data-raw/malvaceae-images/get-tree-1.png" width="90%" style="background-color: #DCDCDC; padding:10px; display: inline-block;"></p>
<hr>
<p>Now, let’s look at some properties of the tree:</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Ntip</a></span><span class="op">(</span><span class="va">original_tree</span><span class="op">)</span>  <span class="co"># gets the number of tips</span>
<span class="co">#&gt; [1] 31</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/root.html">is.rooted</a></span><span class="op">(</span><span class="va">original_tree</span><span class="op">)</span>  <span class="co"># check that it is rooted</span>
<span class="co">#&gt; [1] FALSE</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/is.binary.tree.html">is.binary</a></span><span class="op">(</span><span class="va">original_tree</span><span class="op">)</span>  <span class="co"># check that it is fully resolved</span>
<span class="co">#&gt; [1] FALSE</span>
<span class="fu">datelife</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/datelife/man/phylo_has_brlen.html">phylo_has_brlen</a></span><span class="op">(</span><span class="va">original_tree</span><span class="op">)</span>  <span class="co"># checks that it has branch lengths</span>
<span class="co">#&gt; [1] FALSE</span></pre></div>
<p>The tree has 31 tips, is rooted, has no branch lengths and is not fully resolved, as you probably already noticed. Also, labels correspond to the labels reported on the original study <a href="http://dx.doi.org/10.1600/036364406775971714">here</a>. Other labels are available to use as tip labels. For example, you can plot the tree using other labels, like the unified taxonomic names, or the taxonomic ids.</p>
<div id="ii--getting-the-underlying-alignment" class="section level2">
<h2 class="hasAnchor">
<a href="#ii--getting-the-underlying-alignment" class="anchor"></a>II. Getting the underlying alignment</h2>
<div id="using-physcraper-and-the-arguments--tb-and--no_est" class="section level3">
<h3 class="hasAnchor">
<a href="#using-physcraper-and-the-arguments--tb-and--no_est" class="anchor"></a>Using physcraper and the arguments <code>-tb</code> and <code>-no_est</code>
</h3>
<p>This will download the alignment directly from TreeBASE:</p>
<pre><code>physcraper_run.py -s pg_55 -t tree5864 -tb -no_est -o data/pg_55</code></pre>
<p>The alignment will be saved as <code>pg_55tree5864.aln</code> and <code>physcraper_pg_55tree5864.fas</code> in the inputs folder.</p>
</div>
<div id="downloading-the-alignment-directly-from-a-repository" class="section level3">
<h3 class="hasAnchor">
<a href="#downloading-the-alignment-directly-from-a-repository" class="anchor"></a>Downloading the alignment directly from a repository</h3>
<p>We know that the alignments are in the TreeBASE repo at <a href="https://treebase.org/treebase-web/search/study/matrices.html?id=1361" class="uri">https://treebase.org/treebase-web/search/study/matrices.html?id=1361</a></p>
<p>Let’s save them to a file named <code>TB2:Tr240.nex</code> in the <code>data-raw/alignments</code> folder:</p>
<pre><code>wget -O data-raw/alignments/TB2:Tr240.nex "http://purl.org/phylo/treebase/phylows/tree/TB2:Tr240?format=nexus"</code></pre>
</div>
</div>
<div id="iii--starting-a-physcraper-run" class="section level2">
<h2 class="hasAnchor">
<a href="#iii--starting-a-physcraper-run" class="anchor"></a>III. Starting a physcraper run</h2>
<p>We ran this example using the command:</p>
<pre><code>physcraper_run.py -s pg_55 -t tree5864 -a data-raw/alignments/TB2:Tr240.nex -as nexus -o data/pg_55</code></pre>
</div>
<div id="iv--reading-the-physcraper-results" class="section level2">
<h2 class="hasAnchor">
<a href="#iv--reading-the-physcraper-results" class="anchor"></a>IV. Reading the physcraper results</h2>
<div id="the-physcraper-tag" class="section level3">
<h3 class="hasAnchor">
<a href="#the-physcraper-tag" class="anchor"></a>The physcraper tag</h3>
<p>pg_55tree5864_ndhf</p>
</div>
<div id="input-files" class="section level3">
<h3 class="hasAnchor">
<a href="#input-files" class="anchor"></a>Input files</h3>
<p>Physcraper rewrites input files for a couple reasons: reproducibility, taxon name matching, and taxon reconciliation. It writes the config file down if none was provided. Everything is written into the inputs_tag folder.</p>
</div>
<div id="run-files" class="section level3">
<h3 class="hasAnchor">
<a href="#run-files" class="anchor"></a>Run files</h3>
<p>Files in here are also automatically written down by physcraper.</p>
<p>blast runs, alignments, raxml trees, bootstrap</p>
<p>The trees are reconstructed using RAxML, with tip labels corresponding to local ids (e.g., otu42009, otuPS1) and not taxon names (e.g., <em>Ceiba</em>), nor taxonomic ids (e.g., ott or ncbi). Branch lengths are proportional to relative substitution rates. The RAxML tree with taxon names as tip labels is saved on the <code>outputs_tag</code> folder.</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="va">updated_tree_otus</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"../data/pg_55/run_pg_55tree5864_ndhf/RAxML_bestTree.2020-06-09"</span><span class="op">)</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/plot.phylo.html">plot.phylo</a></span><span class="op">(</span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/ladderize.html">ladderize</a></span><span class="op">(</span><span class="va">updated_tree_otus</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"phylogram"</span>, cex <span class="op">=</span> <span class="fl">0.25</span>, label.offset <span class="op">=</span> <span class="fl">0.001</span>, edge.width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/add.scale.bar.html">add.scale.bar</a></span><span class="op">(</span>cex <span class="op">=</span> <span class="fl">0.3</span>, font <span class="op">=</span> <span class="fl">1</span>, col <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span><span class="st">"Updated tree - otu tags as labels"</span>, side <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></pre></div>
<p><img src="../data-raw/malvaceae-images/unnamed-chunk-7-1.png" width="90%" style="background-color: #6B8E23; padding:10px; display: inline-block;"></p>
</div>
<div id="output-files" class="section level3">
<h3 class="hasAnchor">
<a href="#output-files" class="anchor"></a>Output files</h3>
<p>A number of files is automatically written down by physcraper.</p>
<p>A nexson tree with all types of tip labels is saved in here. From this tree, a tree with any kind of label can be produced. By default, the updtaed tree with taxon names as tip labels is saved in the <code>output_tag</code> folder as <code>updated_taxonname.tre</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="va">updated_tree_taxonname</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span>
  file <span class="op">=</span> <span class="st">"../data/pg_55/outputs_pg_55tree5864_ndhf/updated_taxonname.tre"</span><span class="op">)</span>
<span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label.original</span> <span class="op">&lt;-</span> <span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label</span></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/plot.phylo.html">plot.phylo</a></span><span class="op">(</span><span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/ladderize.html">ladderize</a></span><span class="op">(</span><span class="va">updated_tree_taxonname</span><span class="op">)</span>, type <span class="op">=</span> <span class="st">"phylogram"</span>, cex <span class="op">=</span> <span class="fl">0.2</span>, label.offset <span class="op">=</span> <span class="fl">0.001</span>, edge.width <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/add.scale.bar.html">add.scale.bar</a></span><span class="op">(</span>cex <span class="op">=</span> <span class="fl">0.3</span>, font <span class="op">=</span> <span class="fl">1</span>, col <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/mtext.html">mtext</a></span><span class="op">(</span><span class="st">"Updated tree - Taxon names as labels"</span>, side <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></pre></div>
<p><img src="../data-raw/malvaceae-images/unnamed-chunk-10-1.png" width="90%" style="background-color: #9ecff7; padding:10px; display: inline-block;"></p>
</div>
</div>
<div id="iv--analyzing-the-physcraper-results" class="section level2">
<h2 class="hasAnchor">
<a href="#iv--analyzing-the-physcraper-results" class="anchor"></a>IV. Analyzing the physcraper results</h2>
<p>Compare the original tree with the pruned updated tree</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="va">original_tree_otus</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"../data/pg_55/inputs_pg_55tree5864_ndhf/physcraper_pg_55tree5864_ndhf.tre"</span><span class="op">)</span>
<span class="va">updated_tree_otus_pruned</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span>
  file <span class="op">=</span> <span class="st">"../data/pg_55/pruned_updated.tre"</span>
<span class="op">)</span>
<span class="co"># This tree only has 31 tips because it is the pruned updated tree.</span></pre></div>
<p>Lets get indices from the new tips added to the updated tree and see which taxa they correspond to:</p>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="va">new_tips</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"_otuPS[0-9]*"</span>, <span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label.original</span><span class="op">)</span>
<span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label.original</span><span class="op">[</span><span class="va">new_tips</span><span class="op">]</span>
<span class="co">#&gt;  [1] "Fremontodendron_californicum_otuPS13"       </span>
<span class="co">#&gt;  [2] "Quararibea_costaricensis_otuPS38"           </span>
<span class="co">#&gt;  [3] "Matisia_cordata_otuPS39"                    </span>
<span class="co">#&gt;  [4] "Hibiscus_bojerianus_otuPS45"                </span>
<span class="co">#&gt;  [5] "Macrostelia_laurina_otuPS29"                </span>
<span class="co">#&gt;  [6] "Talipariti_tiliaceum_var._tiliaceum_otuPS48"</span>
<span class="co">#&gt;  [7] "Talipariti_hamabo_otuPS47"                  </span>
<span class="co">#&gt;  [8] "Papuodendron_lepidotum_otuPS46"             </span>
<span class="co">#&gt;  [9] "Cephalohibiscus_peekelii_otuPS34"           </span>
<span class="co">#&gt; [10] "Kokia_kauaiensis_otuPS32"                   </span>
<span class="co">#&gt; [11] "Kokia_drynarioides_otuPS31"                 </span>
<span class="co">#&gt; [12] "Kokia_cookei_otuPS30"                       </span>
<span class="co">#&gt; [13] "Ochroma_pyramidale_otuPS40"                 </span>
<span class="co">#&gt; [14] "Catostemma_fragrans_otuPS37"                </span>
<span class="co">#&gt; [15] "Scleronema_micranthum_otuPS44"              </span>
<span class="co">#&gt; [16] "Cavanillesia_platanifolia_otuPS50"          </span>
<span class="co">#&gt; [17] "Spirotheca_rosea_otuPS42"                   </span>
<span class="co">#&gt; [18] "Bombax_buonopozense_otuPS49"                </span>
<span class="co">#&gt; [19] "Ceiba_acuminata_otuPS36"                    </span>
<span class="co">#&gt; [20] "Ceiba_crispiflora_otuPS41"                  </span>
<span class="co">#&gt; [21] "Pachira_aquatica_otuPS35"                   </span>
<span class="co">#&gt; [22] "Septotheca_tessmannii_otuPS11"              </span>
<span class="co">#&gt; [23] "Triplochiton_zambesiacus_otuPS43"           </span>
<span class="co">#&gt; [24] "Sterculia_tragacantha_otuPS28"</span></pre></div>
<p>There are 24 new tips in the updated tree.</p>
<p>Now plot the trees face to face.</p>
<p>We can first prune the updated tree of the new tips, so it is a straight forward comparison:</p>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="va">cotree</span> <span class="op">&lt;-</span>  <span class="fu">phytools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phytools/man/cophylo.html">cophylo</a></span><span class="op">(</span><span class="va">original_tree_otus</span>, <span class="va">updated_tree_otus_pruned</span>, rotate.multi <span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></pre></div>
<p>Rotating nodes to optimize matching… Done.</p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="fu">phytools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phytools/man/cophylo.html">plot.cophylo</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">cotree</span>, fsize <span class="op">=</span> <span class="fl">0.5</span>, lwd <span class="op">=</span> <span class="fl">0.5</span>, mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.1</span>,<span class="fl">.2</span>,<span class="fl">2</span>,<span class="fl">.3</span><span class="op">)</span>, ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">.1</span>,<span class="fl">1</span><span class="op">)</span>, scale.bar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">.05</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="st">"Original tr.   Pruned updated tr."</span>, cex <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></pre></div>
<p><img src="../data-raw/malvaceae-images/cotree-plot1-1.png" width="90%" style="background-color: #8B008B; padding:10px; display: inline-block;"></p>
<p>But it is more interesting to plot it with all the new tips, so we see exactly where the new things are:</p>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="va">original_tree_taxonname</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/read.tree.html">read.tree</a></span><span class="op">(</span>file <span class="op">=</span> <span class="st">"../data/pg_55/inputs_pg_55tree5864_ndhf/taxonname.tre"</span><span class="op">)</span>
<span class="va">cotree2</span> <span class="op">&lt;-</span>  <span class="fu">phytools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phytools/man/cophylo.html">cophylo</a></span><span class="op">(</span>
  <span class="va">original_tree_taxonname</span>,
  <span class="va">updated_tree_taxonname</span>,
  rotate.multi <span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></pre></div>
<p>Rotating nodes to optimize matching… Done.</p>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="fu">phytools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phytools/man/cophylo.html">plot.cophylo</a></span><span class="op">(</span>
  x <span class="op">=</span> <span class="va">cotree2</span>,
  fsize <span class="op">=</span> <span class="fl">0.3</span>,
  lwd <span class="op">=</span> <span class="fl">0.4</span>,
  mar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">.1</span>,<span class="fl">.1</span>,<span class="fl">2</span>,<span class="fl">.5</span><span class="op">)</span>,
  ylim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">.1</span>,<span class="fl">1</span><span class="op">)</span>,
  scale.bar<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">.05</span><span class="op">)</span>,
  <span class="co"># link.type="curved",</span>
  link.lwd<span class="op">=</span><span class="fl">3</span>,
  link.lty<span class="op">=</span><span class="st">"solid"</span>,
  link.col<span class="op">=</span><span class="fu">phytools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phytools/man/densityTree.html">make.transparent</a></span><span class="op">(</span><span class="st">"#8B008B"</span>,<span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/title.html">title</a></span><span class="op">(</span><span class="st">"Original tree      Updated tree"</span>, cex <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></pre></div>
<p><img src="../data-raw/malvaceae-images/cotree-plot2-1.png" width="90%" style="background-color: #8B008B; padding:10px; display: inline-block;"></p>
<p>We can also plot the updated tree against the synthetic subtree of Malvaceae, to visualize how it updates our current knowledge of the phylogenetic relationships within the family:</p>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="va">tolsubtree</span> <span class="op">&lt;-</span> <span class="fu">rotl</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/rotl/reference/tol_subtree.html">tol_subtree</a></span><span class="op">(</span>ott_id <span class="op">=</span> <span class="fl">279960</span><span class="op">)</span></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Ntip</a></span><span class="op">(</span><span class="va">tolsubtree</span><span class="op">)</span>
<span class="co">#&gt; [1] 5898</span>

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">tolsubtree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span>
<span class="co">#&gt; [1] "Perrierophytum_rubrum_ott3190"       </span>
<span class="co">#&gt; [2] "Perrierophytum_humbertii_ott3195"    </span>
<span class="co">#&gt; [3] "Kosteletzkya_reflexiflora_ott524595" </span>
<span class="co">#&gt; [4] "Kosteletzkya_velutina_ott524593"     </span>
<span class="co">#&gt; [5] "Perrierophytum_glomeratum_ott6118239"</span>
<span class="co">#&gt; [6] "Perrierophytum_hispidum_ott6118240"</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span>
<span class="co">#&gt; [1] "Fremontodendron_californicum_otuPS13"       </span>
<span class="co">#&gt; [2] "Quararibea_costaricensis_otuPS38"           </span>
<span class="co">#&gt; [3] "Matisia_cordata_otuPS39"                    </span>
<span class="co">#&gt; [4] "Hibiscus_bojerianus_otuPS45"                </span>
<span class="co">#&gt; [5] "Macrostelia_laurina_otuPS29"                </span>
<span class="co">#&gt; [6] "Talipariti_tiliaceum_var._tiliaceum_otuPS48"</span></pre></div>
<p><code><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">ape::Ntip</a></code> tells us that there are almost 6k taxa in the Malvaceae OpenTree synthetic tree.</p>
<p>Tip labels in both trees have taxon identifier tags that will make name matching impossible. Lets remove them:</p>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_otuPS[0-9]*"</span>, <span class="st">""</span>, <span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label.original</span><span class="op">)</span>
<span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_otu[0-9]*"</span>, <span class="st">""</span>, <span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span>

<span class="va">tolsubtree</span><span class="op">$</span><span class="va">tip.label.original</span> <span class="op">&lt;-</span> <span class="va">tolsubtree</span><span class="op">$</span><span class="va">tip.label</span>
<span class="va">tolsubtree</span><span class="op">$</span><span class="va">tip.label</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_ott[0-9]*"</span>, <span class="st">""</span>, <span class="va">tolsubtree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span></pre></div>
<p>Eliminate ott and otu tags before comparison, otherwise nothing matches</p>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="co"># cotree3 &lt;-  phytools::cophylo(tolsubtree, updated_tree_taxonname, rotate.multi =TRUE)</span>
<span class="co"># phytools::plot.cophylo(x = cotree3, fsize = 0.2, lwd = 0.5, mar=c(.1,.1,2,.3), ylim=c(-.1,1), scale.bar=c(0, .05))</span>
<span class="co"># title("Tol subtree      Updated tree", cex = 0.1)</span></pre></div>
<p>The whole Malvaceae tree might be too large for a cotree plot. Lets try something else.</p>
<p>First, get indices of new tips from updated tree present in the Malvaceae synthetic tree:</p>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="va">all_updated_tree_in_tol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label</span>, <span class="va">tolsubtree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span>
<span class="va">new_tips_in_tol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label</span><span class="op">[</span><span class="va">new_tips</span><span class="op">]</span>, <span class="va">tolsubtree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span>
<span class="co"># one tip seems to be absent frm the Malvaceae subtree</span>
<span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label</span><span class="op">[</span><span class="fl">47</span><span class="op">]</span>
<span class="co">#&gt; [1] "Firmiana_platanifolia"</span>
<span class="co"># going to OpenTree it looks like it atually corresponds to Firmiana simplex, not sure why it was not standardized, probably bc it was in NCBI</span>
<span class="co"># I'm changing it by hand:</span>
<span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label</span><span class="op">[</span><span class="fl">47</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Firmiana_simplex"</span>
<span class="co"># And running match again</span>
<span class="va">new_in_tol</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">updated_tree_taxonname</span><span class="op">$</span><span class="va">tip.label</span>, <span class="va">tolsubtree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span></pre></div>
<p>Then, use the `phytools package to create a data object that will contain info from our tolsubtree tips and edges:</p>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="va">tolsubtree</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/ladderize.html">ladderize</a></span><span class="op">(</span><span class="va">tolsubtree</span><span class="op">)</span>

<span class="va">tip_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"gray"</span>, <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Ntip</a></span><span class="op">(</span><span class="va">tolsubtree</span><span class="op">)</span><span class="op">)</span>
<span class="va">tip_colors</span><span class="op">[</span><span class="va">all_updated_tree_in_tol</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"black"</span>
<span class="va">tip_colors</span><span class="op">[</span><span class="va">new_tips_in_tol</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"red"</span>

<span class="va">tip_values</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/summary.phylo.html">Ntip</a></span><span class="op">(</span><span class="va">tolsubtree</span><span class="op">)</span><span class="op">)</span>
<span class="va">tip_values</span><span class="op">[</span><span class="va">tip_colors</span> <span class="op">==</span> <span class="st">"black"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">1</span>
<span class="va">tip_values</span><span class="op">[</span><span class="va">tip_colors</span> <span class="op">==</span> <span class="st">"red"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">100</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tip_values</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">tolsubtree</span><span class="op">$</span><span class="va">tip.label</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">tip_values</span><span class="op">)</span>
<span class="co">#&gt;     Perrierophytum_rubrum  Perrierophytum_humbertii Kosteletzkya_reflexiflora </span>
<span class="co">#&gt;                         0                         0                         0 </span>
<span class="co">#&gt;     Kosteletzkya_velutina Perrierophytum_glomeratum   Perrierophytum_hispidum </span>
<span class="co">#&gt;                         0                         0                         0</span>
<span class="va">all_values</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/sum_tips.html">sum_tips</a></span><span class="op">(</span><span class="va">tolsubtree</span>, <span class="va">tip_values</span><span class="op">)</span>

<span class="va">all_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"gray"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">all_values</span><span class="op">)</span><span class="op">)</span>
<span class="va">all_values</span><span class="op">[</span><span class="va">all_values</span> <span class="op">&gt;=</span> <span class="fl">100</span><span class="op">]</span>
<span class="co">#&gt;   [1]  100  100  100  100  100  100  100  100  100  100  100  100  100  100  100</span>
<span class="co">#&gt;  [16]  100  100  100  100  100  100  100  100  100 2422 2413 2413 2413 2413 2413</span>
<span class="co">#&gt;  [31] 2413 2313 2213 1213 1013  900  700  700  700  400  100  100  300  300  200</span>
<span class="co">#&gt;  [46]  200  200  100  100  100  100  100  100  100  100  100  100  300  300  300</span>
<span class="co">#&gt;  [61]  300  300  200  200  100  113  109  109  105  101  100  100  100 1000  800</span>
<span class="co">#&gt;  [76]  800  300  300  200  100  100  100  500  100  100  400  200  200  100  100</span>
<span class="co">#&gt;  [91]  200  200  100  100  100  200  100  100  100  100  100  100  100  100</span>
<span class="va">all_values</span><span class="op">[</span><span class="va">all_values</span> <span class="op">&gt;=</span> <span class="fl">1</span><span class="op">]</span>
<span class="co">#&gt;   [1]  100  100  100  100  100  100  100  100  100    1  100    1    1    1    1</span>
<span class="co">#&gt;  [16]    1    1    1    1    1    1    1    1  100  100  100  100  100  100  100</span>
<span class="co">#&gt;  [31]  100  100  100  100  100  100  100    1    1    1    1    1    1    1    1</span>
<span class="co">#&gt;  [46]    1 2422 2413 2413 2413 2413 2413 2413 2313 2213 1213 1013  900  700  700</span>
<span class="co">#&gt;  [61]  700  400  100  100  300  300  200  200  200  100  100  100  100  100  100</span>
<span class="co">#&gt;  [76]  100  100  100  100  300  300  300  300  300  200  200  100  113  109  109</span>
<span class="co">#&gt;  [91]  105  101    1  100  100    4    4    1    4    3    2    2    1    1  100</span>
<span class="co">#&gt; [106] 1000  800  800  300  300  200  100  100  100  500  100  100  400  200  200</span>
<span class="co">#&gt; [121]  100  100  200  200  100  100  100  200  100  100  100  100  100  100  100</span>
<span class="co">#&gt; [136]  100    1    2    2    1    1</span>

<span class="va">all_cols</span><span class="op">[</span><span class="va">all_values</span> <span class="op">&gt;=</span> <span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"black"</span>
<span class="va">all_cols</span><span class="op">[</span><span class="va">all_values</span> <span class="op">&gt;=</span> <span class="fl">100</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"red"</span>
<span class="va">edge_colors</span> <span class="op">&lt;-</span> <span class="va">all_cols</span><span class="op">[</span><span class="va">tolsubtree</span><span class="op">$</span><span class="va">edge</span><span class="op">[</span>,<span class="fl">2</span><span class="op">]</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">edge_colors</span><span class="op">)</span> <span class="op">==</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span><span class="op">(</span><span class="va">tolsubtree</span><span class="op">$</span><span class="va">edge</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="co"># tip_cols &lt;- all_cols[seq(ape::Ntip(tolsubtree))]</span>

<span class="va">namess</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"Synth tips"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">edge_colors</span><span class="op">)</span><span class="op">)</span>
<span class="va">namess</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">edge_colors</span> <span class="op">==</span> <span class="st">"black"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Original tips"</span>
<span class="va">namess</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">edge_colors</span> <span class="op">==</span> <span class="st">"red"</span><span class="op">)</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"Updated tips"</span>

<span class="va">tolsubtree</span><span class="op">$</span><span class="va">maps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">as.list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">edge_colors</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
<span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">edge_colors</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">tolsubtree</span><span class="op">$</span><span class="va">maps</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">namess</span><span class="op">[</span><span class="va">i</span><span class="op">]</span>
<span class="op">}</span>
<span class="co"># fam_tree_brlen3$maps[1]</span>
<span class="co"># str(fam_tree_brlen3$maps)</span>
<span class="va">cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/setNames.html">setNames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>,<span class="st">"black"</span>, <span class="st">"gray"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Updated tips"</span>, <span class="st">"Original tips"</span>, <span class="st">"Synth tips"</span><span class="op">)</span><span class="op">)</span>
<span class="va">tolsubtree_wbl</span> <span class="op">&lt;-</span> <span class="fu">ape</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ape/man/compute.brlen.html">compute.brlen</a></span><span class="op">(</span><span class="va">tolsubtree</span><span class="op">)</span>
<span class="va">tolsubtree_wbl</span><span class="op">$</span><span class="va">maps</span> <span class="op">&lt;-</span> <span class="va">tolsubtree</span><span class="op">$</span><span class="va">maps</span></pre></div>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>fg<span class="op">=</span><span class="st">"transparent"</span><span class="op">)</span>
<span class="fu">phytools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phytools/man/plotSimmap.html">plotSimmap</a></span><span class="op">(</span><span class="va">tolsubtree_wbl</span>,type<span class="op">=</span><span class="st">"fan"</span>,part<span class="op">=</span><span class="fl">0.95</span>, ftype<span class="op">=</span><span class="st">"i"</span>, colors <span class="op">=</span> <span class="va">cols</span>, lwd <span class="op">=</span> <span class="fl">0.1</span>, fsize <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span> <span class="co">#lwd = 0.5, fsize=0.1</span>
<span class="va">lastPP</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/get.html">get</a></span><span class="op">(</span><span class="st">"last_plot.phylo"</span>,env<span class="op">=</span><span class="va">.PlotPhyloEnv</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>fg<span class="op">=</span><span class="st">"black"</span><span class="op">)</span>
<span class="fu">phytools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phytools/man/add.simmap.legend.html">add.simmap.legend</a></span><span class="op">(</span>colors<span class="op">=</span><span class="va">cols</span>,vertical<span class="op">=</span><span class="cn">FALSE</span>,x<span class="op">=</span><span class="op">-</span><span class="fl">0.9</span>, y<span class="op">=</span><span class="op">-</span><span class="fl">1.2</span>,prompt<span class="op">=</span><span class="cn">FALSE</span>, fsize <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>
<span class="va">tt</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"_"</span>,<span class="st">" "</span>,<span class="va">tolsubtree_wbl</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span>
<span class="co">#tt[tip_colors == "gray"] &lt;- ""</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/text.html">text</a></span><span class="op">(</span><span class="va">lastPP</span><span class="op">$</span><span class="va">xx</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">tt</span><span class="op">)</span><span class="op">]</span>,<span class="va">lastPP</span><span class="op">$</span><span class="va">yy</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">tt</span><span class="op">)</span><span class="op">]</span>,
  <span class="va">tt</span>,cex<span class="op">=</span><span class="fl">0.2</span>,col<span class="op">=</span><span class="va">tip_colors</span>,pos<span class="op">=</span><span class="fl">4</span>,offset<span class="op">=</span><span class="fl">0.1</span>,font<span class="op">=</span><span class="fl">3</span><span class="op">)</span></pre></div>
<p><img src="../data-raw/malvaceae-images/simmap-plot-1.png" width="90%" style="background-color: #8B008B; padding:10px; display: inline-block;"></p>
</div>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<p>Trick for the cophylo titles and margins from <a href="https://cran.r-project.org/web/packages/phangorn/vignettes/IntertwiningTreesAndNetworks.html" class="uri">https://cran.r-project.org/web/packages/phangorn/vignettes/IntertwiningTreesAndNetworks.html</a></p>
<p>Trick for colord tip labels: <a href="http://blog.phytools.org/2013/09/coloring-tip-labels-by-character-state.html" class="uri">http://blog.phytools.org/2013/09/coloring-tip-labels-by-character-state.html</a></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Luna L. Sanchez Reyes, Emily Jane McTavish.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
